Add inventory withChildren interceptor for backward compatibility (#4527)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1024-0-0-explicit-withchildren-parameter-for-inventory-api-calls-announcement.md

---

> date: '2026-04-08'

Reviewer: ```suggestion
date: 
```

---
Starting with version 1024.0.0, the {{< product-c8y-iot >}} platform changes the server-side default of the `withChildren` query parameter on `/inventory/managedObjects` REST endpoints from `true` to `false` for performance optimization. The entire UI codebase has been adapted to be compatible with the new default.

Two categories of changes were made:

* **AngularJS (legacy modules)**: A new HTTP interceptor `c8yInventoryWithChildrenInterceptor` was added. It automatically appends `withChildren=true` to all GET requests to `/inventory/managedObjects` endpoints (list and detail, including `/childAssets` and `/childDevices` subresources) if the parameter is not already set. This preserves the old behavior for all legacy AngularJS code without requiring individual call-site changes.
* **Angular (@c8y/ngx-components)**: Every `InventoryService.detail()`, `InventoryService.list()`, and `InventoryService.childAssetsList()` call across `@c8y/ngx-components` was updated to explicitly pass `withChildren: false` where child references are not needed, or `withChildren: true` where they are. This makes each call site's intent explicit rather than relying on the server default.

> **Breaking change**

Reviewer 2: @luja-c8y is this really a breaking change on the Web SDK? Isn't this only breaking if the BE team change it and then they need to do this announcement? Or was there really a change in the InventoryService?

SDK consumers who call `InventoryService.detail()`, `.list()`, `.listQuery()`, `.childAssetsList()`, or `.childDevicesList()` without specifying `withChildren` are affected. Previously, the server returned child references by default; now it does not. Responses have smaller payloads but are missing `childAssets`, `childDevices`, and `childAdditions` references unless explicitly requested.

AngularJS plugin developers are shielded by the interceptor and their existing code continues to work as before.

> **Migration**: All `InventoryService` calls that need child references in the response must now explicitly pass `withChildren: true`. Calls that do not need child references should pass `withChildren: false` or omit the parameter, which now defaults to `false` on the server. Review all `inventoryService.detail()` and `inventoryService.list()` call sites in custom code.

Reviewer 3: ```suggestion
Starting with version 1024.0.0, the {{< product-c8y-iot >}} platform changes the server-side
default of the `withChildren` query parameter on `/inventory/managedObjects` GET endpoints from
`true` to `false` for performance optimization. Responses no longer include the `childAssets`,
`childDevices`, and `childAdditions` reference arrays unless `withChildren=true` is set.

This is a server-side change. The Web SDK's `InventoryService` is unchanged — it passes
`withChildren` straight through and never set a default. The UI codebase has been adapted:

* **AngularJS (legacy modules)**: A new HTTP interceptor `c8yInventoryWithChildrenInterceptor`
  appends `withChildren=true` to `/inventory/managedObjects` GET requests when the parameter is
  not already set, preserving existing behavior without call-site changes.
* **Angular (`@c8y/ngx-components`)**: Inventory call sites now pass `withChildren` explicitly.
  Shipped components behave correctly under the new default; no action is required.

**Migration**

If your custom code reads `childAssets`, `childDevices`, or `childAdditions` from an
`InventoryService.detail()`, `.list()`, or `.listQuery()` response, add `withChildren: true` to
those calls. Otherwise those references will be missing once the new default is active for the
tenant.
```

---
date: '2026-04-08'

> title: Explicit withChildren parameter for inventory API calls

Reviewer: ```suggestion
title: Explicit withChildren parameter for Inventory API calls
```

product_area: Application enablement & solutions
change_type:
  - value: change-inv-3bw8e
    label: Announcement
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-64714
version: 1024.0.0
environment_availability:

>   - label: eu-latest.cumulocity.com

Reviewer: Is this change already available in all those environments? Otherwise, leave them empty as they will be filled in automatically.

Author: No.. I was pretty sure this is a "standard" part of a header. I've changed it as you suggested.

> Starting with version 1024.0.0, the {{< product-c8y-iot >}} platform changes the server-side

Reviewer 2: I guess BE is versioned differently. Maybe jsut saying that it will be soon changed and the default is deprecated is enough?

Reviewer 3: update, please check if it's ok.

```
The {{< product-c8y-iot >}} platform will soon change the server-side default of the
`withChildren` query parameter on `/inventory/managedObjects` GET endpoints from `true` to `false`
for performance optimization; relying on the current `true` default is deprecated. Once the new
default takes effect, responses no longer include the `childAssets`, `childDevices`, and
`childAdditions` reference arrays unless `withChildren=true` is set explicitly.
```

default of the `withChildren` query parameter on `/inventory/managedObjects` GET endpoints from
`true` to `false` for performance optimization. Responses no longer include the `childAssets`,

> `childDevices`, and `childAdditions` reference arrays unless `withChildren=true` is set.

Reviewer 3: ```suggestion
The {{< product-c8y-iot >}} platform will soon change the server-side default of the
`withChildren` query parameter on `/inventory/managedObjects` GET endpoints from `true` to `false`
for performance optimization; relying on the current `true` default is deprecated. Once the new
default takes effect, responses no longer include the `childAssets`, `childDevices`, and
`childAdditions` reference arrays unless `withChildren=true` is set explicitly.
```

  - label: cumulocity.com
---
Starting with version 1024.0.0, the {{< product-c8y-iot >}} platform changes the server-side default of the `withChildren` query parameter on `/inventory/managedObjects` REST endpoints from `true` to `false` for performance optimization. The entire UI codebase has been adapted to be compatible with the new default.

Two categories of changes were made:

> * **AngularJS (legacy modules)**: A new HTTP interceptor `c8yInventoryWithChildrenInterceptor` was added. It automatically appends `withChildren=true` to all GET requests to `/inventory/managedObjects` endpoints (list and detail, including `/childAssets` and `/childDevices` sub-resources) if the parameter is not already set. This preserves the old behavior for all legacy AngularJS code without requiring individual call-site changes.

Reviewer: ```suggestion
* **AngularJS (legacy modules)**: A new HTTP interceptor `c8yInventoryWithChildrenInterceptor` was added. It automatically appends `withChildren=true` to all GET requests to `/inventory/managedObjects` endpoints (list and detail, including `/childAssets` and `/childDevices` subresources) if the parameter is not already set. This preserves the old behavior for all legacy AngularJS code without requiring individual call-site changes.
```
